[10.0][FIX] Several very important fixes

alexis-via opened the pull request:
Here is the scenario of the bug:
1) create a customer invoice with "Intrastat Transaction Type" = False and validate it
2) click on the "Refund" button and follow the wizard to create a draft refund
3) In the draft refund created, "Intrastat Transaction Type" has the default value for customer invoices, instead of having the default value for customer refunds

Code analysis: when you look at the code of the refund wizard, you see that it doesn't set the 'type' in the context, so the value of 'type' in the context is still 'out_invoice', so the default value for "Intrastat Transaction Type" is the one of Customer Invoices.

For me, the solution is just to not put any default value for the field "Intrastat Transaction Type". The code that generate the transaction lines of the intrastat declaration will use the default value for invoices where the field ""Intrastat Transaction Type" is empty.

alexis-via committed:
[FIX] Intrastat Transaction Type on refunds created from invoices

pedrobaeza commented:
I think it's better to overload the method that creates the refund and force the value, don't you think?

alexis-via commented:
If we do that, the problem will happen everytime the 'type' is not properly set in the context, which may happen at several other places.
@luc-demeyer What's your point of view on this ?

alexis-via committed:
[FIX] multi-company issues

alexis-via commented:
I added compute_sudo=True for the fields that use information from partners and store it on invoices, to fix multi-company problems.

alexis-via committed:
FIX PEP8 warnings

alexis-via committed:
[FIX] Region on arrival intrastat declaration

alexis-via commented:
@luc-demeyer I found one more very important bug: the computation of region was wrong on arrival intrastat declaration: we were taking source location instead of destination location on the move. I'm very surprised we didn't find it earlier ; it seems it was already there on v8. It's one of my customers who found it.

alexis-via commented:
I backported the fix on region to v8 here: https://github.com/OCA/intrastat/pull/36
In fact, I was not impacted by the bug on v8 because I don't use _get_region in l10n_fr_intrastat_product, which explains why my users were not affected.

luc-demeyer committed:
fix message_post

while reviewing this PR, I found a bug in the message_post calls. This PR fixes it.

luc-demeyer commented:
I have performed code review and tested this PR.
During the test I found a bug that is unrelated to this PR but since the fix is straightforward I propose to merge it into this PR (cf. https://github.com/akretion/intrastat/pull/1) since this PR already combines a couple of fixes.
After merging https://github.com/akretion/intrastat/pull/1 into this PR, this PR can be merged.

Commit:
Merge pull request #1 from luc-demeyer/patch-6

fix message_post

alexis-via commented:
@pedrobaeza Could you add write access to @luc-demeyer on this repo ? Github says "At least 1 approving review is required by reviewers with write access" and luc approved, but he is not counted.

alexis-via commented:
and by the way, I think it's not fair to count only people with write access, because there are too few of them and this project has a very small audience.
